web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,161 @@
+
+The second problem is that the conventions for constructing types from a given type backend are inconsistent. Let's review string aliases used to construct certain types:
+
+| logical type | NumPy   | pandas extension | pyarrow                  |
+| int          | "int64" | "Int64"          | "int64[pyarrow]"         |
+| string       | N/A     | "string"         | N/A                      |
+| datetime     | N/A     | "datetime64[us]" | "timestamp[us][pyarrow]" |
+
+"string[pyarrow]" is excluded from the above table because it is misleading; while "int64[pyarrow]" definitely gives you a pyarrow backed string, "string[pyarrow]" gives you a pandas extension array which itself then uses pyarrow, which can introduce behavior differences (see [issue 58321](https://github.com/pandas-dev/pandas/issues/58321)).
+
+If you wanted to try and be more explicit about using pyarrow, you could use the ``pd.ArrowDtype`` wrapper. But this unfortunately exposes gaps when trying to use that pattern across all backends:
+
+| logical type | NumPy    | pandas extension | pyarrow                           |
+| int          | np.int64 | pd.Int64Dtype()  | pd.ArrowDtype(pa.int64())         |
+| string       | N/A      | pd.StringDtype() | pd.ArrowDtype(pa.string())        |
+| datetime     | N/A      | ???              | pd.ArrowDtype(pa.timestamp("us")) |
+
+It would stand to reason in this approach that you could use a ``pd.DatetimeDtype()`` but no such type exists (there is a ``pd.DatetimeTZDtype`` which requires a timezone).
+
+The third issue is that the extent to which pandas may support any given type is unclear. Issue [#58307](https://github.com/pandas-dev/pandas/issues/58307) highlights one example. It would stand to reason that you could interchangeably use a pandas datetime64 and a pyarrow timestamp, but that is not always true. Another common example is the use of NumPy fixed length strings, which users commonly try to use even though we claim no real support for them (see [#5764](https://github.com/pandas-dev/pandas/issues/57645)).
+
+## Assessing the Current Type System(s)
+
+A best effort at visualizing the current type system(s) with types that we currently "support" or reasonably may want to is shown [in this comment](https://github.com/pandas-dev/pandas/issues/58141#issuecomment-2047763186). Note that this does not include the ["pyarrow_numpy"](https://github.com/pandas-dev/pandas/pull/58451) string data type or the string data type that uses the NumPy 2.0 variable length string data type (see [comment](https://github.com/pandas-dev/pandas/issues/57073#issuecomment-2080798081)) as they are under active discussion.
+
+## Proposal
+
+Derived from the hierarchical visual in the previous section, this PDEP proposes that pandas supports at least all of the following _logical_ types, excluding any type widths for brevity:
```

Comment:
IIUC we already support all of these, just not with all three of the storages.  Is this paragraph suggesting that we implement all of these for all three (four if we include sparse) storages?

Reply:
Not at all - this is just suggesting that these be the logical types we offer, whether through the factory functions or class-based design that follows.

In today's world it is the responsibility of the user to know what may or may not be offered by the various backends and pick from them accordingly. With dates/datetimes being a good example, a user would have to explicitly pick from some combination of `pd.ArrowDtype(pa.date32())`, `pd.ArrowDtype(pa.timestamp(<unit>))` and `datetime64[unit]` since we don't offer any logical abstraction. 

In the model of this PDEP, we would offer `pd.date()` and `pd.datetime()` or something similar, abstracting away the type implementation details